gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -623,7 +625,9 @@ protected GoogleCloudStorageItemInfo getInitialMetadata() {
 
       @Override
       protected Storage.Objects.Get createRequest() throws IOException {
-        return configureRequest(super.createRequest(), resourceId.getBucketName());
+        Storage.Objects.Get getObject = super.createRequest();
+        setEncryptionHeaders(getObject);
+        return configureRequest(getObject, resourceId.getBucketName());
```

Comment:
Could it be done in `configureRequest` method? Seems like it  could be a good feet if it applies to all requests.

Reply:
It actually doesn't necessarily apply to all requests. for example, list or delete requests don't required those encryption headers to show the result. Only get/insert/rewrite/open need these encryption headers.

Reply:
Should we use if condition to distinguish these requests in the configuration method then? It's still could be easier to implement if all logic will be in one place?  Something like `request instanceof Insert`.

Reply:
Done.